### PR TITLE
MM-32351 Add Compliance Subsections 

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Adding the sub sections for Compliance section in system console

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32351

#### RelatedPRs
https://github.com/mattermost/mattermost-webapp/pull/7596
https://github.com/mattermost/mattermost-server/pull/17023

### CODE REVIEW

**Review by BenCookie95 — APPROVED** [review]

File: src/constants/permissions.ts [hunk 1]
```diff
@@ -109,8 +109,14 @@ const values = {
     SYSCONSOLE_WRITE_PLUGINS: 'sysconsole_write_plugins',
     SYSCONSOLE_READ_INTEGRATIONS: 'sysconsole_read_integrations',
     SYSCONSOLE_WRITE_INTEGRATIONS: 'sysconsole_write_integrations',
-    SYSCONSOLE_READ_COMPLIANCE: 'sysconsole_read_compliance',
-    SYSCONSOLE_WRITE_COMPLIANCE: 'sysconsole_write_compliance',
+    SYSCONSOLE_READ_COMPLIANCE_DATA_RETENTION: 'sysconsole_read_compliance_data_retention',
+    SYSCONSOLE_WRITE_COMPLIANCE_DATA_RETENTION: 'sysconsole_write_compliance_data_retention',
```
Comment (hahmadia) [inline src/constants/permissions.ts:113]: Nit:
Add `_policy` to the end here and update permission in backend accordingly.
```suggestion
    SYSCONSOLE_READ_COMPLIANCE_DATA_RETENTION_POLICY: 'sysconsole_read_compliance_data_retention_policy',
    SYSCONSOLE_WRITE_COMPLIANCE_DATA_RETENTION_POLICY: 'sysconsole_write_compliance_data_retention_policy',
```

File: src/constants/permissions_sysconsole.ts [hunk 2]
```diff
@@ -11,6 +11,12 @@ export const RESOURCE_KEYS = {
         PERMISSIONS: 'user_management.permissions',
         SYSTEM_ROLES: 'user_management.system_roles',
     },
+    COMPLIANCE: {
+        DATA_RETENTION: 'compliance.data_retention_policy',
```
Comment (hahmadia) [inline src/constants/permissions_sysconsole.ts:15]: Nit:
```suggestion
        DATA_RETENTION_POLICY: 'compliance.data_retention_policy',
```

**Review by hahmadia — COMMENTED** [review]
Some very minor changes.
Also, I think we can remove QA Review from this. Just ensure that your webapp redux hash has the latest commit hash from this PR's commits.
Thanks

**Review by hahmadia — APPROVED** [review]
LGTM!
Please resolve conflicts with new ancillary permission changes! Thanks

**Review by BenCookie95 — APPROVED** [review]

**Pull request merged**